### PR TITLE
examples: Add tcpconnect as a libbpfgo example

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
$ make -C selftest/tcpconnect
$ make -C selftest/tcpconnect run

will build:

- tcpconnect-static
- tcpconnect-dynamic
- tcpconnect-go-static
- tcpconnect-go-dynamic

as an example on how to use libbpfgo (or libbpf) to probe kernel
function "tcp_connection" and get data from it. It will also use
this example as a selftest for a simple TCP connect().

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
This is the old https://github.com/aquasecurity/libbpfgo/pull/38 which I accidentally lost after a force push in my local repository. I'm bringing it back now. 

@grantseltzer thanks for reviewing this, it was missing a Readme.MD file only.

File: selftest/tcpconnect/tcpclient [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+sleep 5 ; echo "connected" | nc -q 2 127.0.0.1 12345 > /dev/null
```
Comment (grantseltzer) [inline selftest/tcpconnect/tcpclient:3]: On fedora the `nc` that's packaged does not have a `-q` flag. The only similar options I see:

`--send-only            Only send data, ignoring received; quit on EOF`
`w, --wait <time>          Connect timeout`

File: selftest/tcpconnect/tcpserver [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+nc -F -l -p 12345 > /dev/null
```
Comment (grantseltzer) [inline selftest/tcpconnect/tcpserver:3]: Same with `-F`. Looks like there's an openbsd, gnu, and 'classic' netcats. A licensing issue.

File: selftest/tcpconnect/tcpclient (hunk 1 above)
Comment (rafaeldtinoco) [reply selftest/tcpconnect/tcpclient:3]: Hum, I'll change that usage then... did not expect that.

File: selftest/tcpconnect/tcpserver (hunk 2 above)
Comment (rafaeldtinoco) [reply selftest/tcpconnect/tcpserver:3]: Same here.

**Comment by rafaeldtinoco** [comment]
@grantseltzer thanks. I believe those are fixed for Fedora now.

**Review by grantseltzer — APPROVED** [review]
LGTM :-)

File: selftest/tcpconnect/README.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+        struct in6_addr daddr6; // destination address (IPv6)
+        u8   thesource;         // I am the one originating packet
+};
+```
+
+This means that both, the eBPF program, and the userland code, need to know about this structure, so they can communicate.
+
+> Note: there are different types of events which an eBPF program can be attached to (links) and different types of maps that can be used to exchange data. I'm not entering that topic here and this example uses simple ones.
+
+From the snippet bellow:
```
Comment (grantseltzer) [inline selftest/tcpconnect/README.md:63]: here too

**Review by grantseltzer — APPROVED** [review]

**Pull request merged**